pkg/controllers/termination/v1alpha1/terminate.go

```diff
@@ -139,13 +123,15 @@ func (t *Terminator) deleteNodes(ctx context.Context, nodes []*v1.Node, provisio
 	return nil
 }
 
-// getNodes returns a list of nodes with the provisioner's labels and given labels
-func (t *Terminator) getNodes(ctx context.Context, provisioner *v1alpha1.Provisioner, additionalLabels map[string]string) ([]*v1.Node, error) {
+// getLabeledNodes returns a list of nodes with the provisioner's labels and given labels
+func (t *Terminator) getLabeledNodes(ctx context.Context, phaseLabel string) ([]*v1.Node, error) {
 	nodes := &v1.NodeList{}
-	if err := t.kubeClient.List(ctx, nodes, client.MatchingLabels(functional.UnionStringMaps(map[string]string{
-		v1alpha1.ProvisionerNameLabelKey:      provisioner.Name,
-		v1alpha1.ProvisionerNamespaceLabelKey: provisioner.Namespace,
-	}, additionalLabels))); err != nil {
+	if err := t.kubeClient.List(ctx, nodes, client.HasLabels([]string{
+		provisioning.ProvisionerNameLabelKey,
+		provisioning.ProvisionerNamespaceLabelKey,
+	}), client.MatchingLabels(map[string]string{
```

Comment:
I'm not sure I'm grokking client.HasLabels and client.MatchingLabels. Why did this change? What's the logic here?

Reply:
Ah gotcha. This is kind of a cool trick. It'll get deleted, but a good tool to have in your back pocket.